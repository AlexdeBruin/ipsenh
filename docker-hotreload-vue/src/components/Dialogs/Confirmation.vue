<template>
  <v-dialog v-model="dialog" max-width="500px">
    <v-card>
      <v-card-title class="headline">
        <template v-if="!loading">{{title}}</template>
        <template v-if="loading">
          <v-progress-circular indeterminate color="red"></v-progress-circular>
          &nbsp;{{loadingText}}
        </template>
      </v-card-title>

      <v-card-actions>
        <v-spacer></v-spacer>

        <v-btn @click="dialogAction()" color="green darken-1" flat="flat">{{buttonText}}</v-btn>

        <v-btn id="cancleButton" color="red darken-1" flat="flat" @click="toggle()">Annuleren</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: "Confirmation",
  data: function() {
    return {
      loading: false,
      dialog: this.showDialog
    };
  },
  props: {
    // Function must return a Promise in order to work.
    method: { type: Function },
    buttonText: {
      type: String,
      required: true
    },
    title: {
      type: String,
      required: true
    },
    loadingText: {
      type: String,
      required: true
    },
    showDialog: {
      type: Boolean,
      default: false,
      required: true
    }
  },
  methods: {
    toggle() {
      this.dialog = false;
      this.$emit("toggle-dialog");
    },
    dialogAction() {
      this.loading = true;
      this.method()
        .then(() => {
          this.loading = false;
          this.toggle();
        })
        .catch(() => {
          this.loading = false;
          this.toggle();
        });
    }
  },
  watch: {
    showDialog: function(newVal, oldVal) {
      this.dialog = newVal;
    }
  }
};
</script>

<style lang="scss" scoped>
</style>