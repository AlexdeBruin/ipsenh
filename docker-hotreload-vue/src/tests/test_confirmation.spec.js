import Confirmation from '../components/Dialogs/Confirmation.vue'
import {mount} from '@vue/test-utils'

const wrapper = mount(Confirmation)

describe('Confirmation', () => {
    test('is a vue instance', () => {
        expect(wrapper.isVueInstance()).toBeTruthy()
    })
    it('Contains a button', () => {
        expect(wrapper.contains('v-btn')).toBe(true)
    })
    it('it can click the cancle button', () => {
        wrapper.vm.dialog = true
        expect(wrapper.vm.dialog).toBe(true)
        const button = wrapper.find('#cancleButton')
        button.trigger('click')
        expect(wrapper.vm.dialog).toBe(false)
    })
    test('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
})