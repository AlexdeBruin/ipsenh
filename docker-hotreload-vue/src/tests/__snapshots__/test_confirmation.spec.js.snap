// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation renders correctly 1`] = `
<v-dialog
  max-width="500px"
>
  <v-card>
    <v-card-title
      class="headline"
    >
       
      <!---->
    </v-card-title>
     
    <v-card-actions>
      <v-spacer />
       
      <v-btn
        color="green darken-1"
        flat="flat"
      >
        
      </v-btn>
       
      <v-btn
        color="red darken-1"
        flat="flat"
        id="cancleButton"
      >
        Annuleren
      </v-btn>
    </v-card-actions>
  </v-card>
</v-dialog>
`;
